spec/helpers/error_summary_helper_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+describe ErrorSummaryHelper do
+  describe '#field_anchor_link' do
+    context 'given a ActiveModel record and a field_name' do
+      let(:model_name_mock) { double(param_key: 'active_model') }
+      let(:active_model_mock) { double(model_name: model_name_mock) }
```

Comment:
You say in the `context` that this is an `ActiveModel record`, but it's a duck — a mock duck! — quacking like one. Why not use the real thing? As a bonus, the spec will then break usefully if the interface changes.

Reply:
I mocked out of habit (since this is a unit spec and that's a collaborator), but I think you're right.

After reading around it a bit ("[don't mock what you don't own](https://github.com/testdouble/contributing-tests/wiki/Don't-mock-what-you-don't-own)"), I see that I don't really get any value from mocking here.

Reply:
Addressed in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/112/commits/98b95042bc27b1476159bae1a21493a025147be7 - keen to get feedback about how clear the inline class declaration is.

Reply:
I've had a chance to mull this over a bit.

1. I felt uneasy about the mocking for the same reason you weren't sure about it: the chaining. I don't think mocking is necessarily wrong here.
1. Calling `model.model_name.param_key` is a Law of Demeter violation. .I know LoD violations aren't always fixable in a nice way but it's usually worth talking about when we do break it.
1. The interface we care about in this method is the one that implements `param_key`.
1. One simplification is to pass in the thing that implements `param_key`, ie the `ActiveModel::Name` itself (https://api.rubyonrails.org/classes/ActiveModel/Name.html) instead of passing the model itself. That makes this method dumber, and the call becomes more explicit.

The view code would then be

```
<a href='<%= field_anchor_link(record.model_name, field) %>'>
```

Or even with kwargs, which I generally favour:

```
<a href='<%= field_anchor_link(model_name: record.model_name, field_name: field) %>'>
```

The spec could then indeed use a mock and wouldn't read so oddly.

What do you think?

Reply:
Had a conversation irl about this, then (hopefully!) resolved it in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/112/commits/d622cdd1379e54d3f7b31b438bc6a86c013249bd.

Reply:
Oh, excuse me, I missed the comment on kwargs, just fixing that...